src/miral/sticky_keys.cpp

```diff
@@ -0,0 +1,185 @@
+{
+public:
+    explicit Self(bool enabled)
+        : state{State{enabled}}
+    {
+    }
+
+    void set_accessibility_manager(std::shared_ptr<msh::AccessibilityManager> const& manager)
+    {
+        auto const locked_state = state.lock();
+        accessibility_manager = manager;
+        manager->sticky_keys_enabled(locked_state->enabled);
+        manager->sticky_keys().should_disable_if_two_keys_are_pressed_together(
+            locked_state->should_disable_if_two_keys_are_pressed_together);
+        manager->sticky_keys().on_modifier_clicked(std::move(locked_state->on_modifier_clicked));
+    }
+
+    void enable()
+    {
+        auto const locked_state = state.lock();
+        if (locked_state->enabled)
+            return;
+
+        locked_state->enabled = true;
+        if (auto const manager = accessibility_manager.lock())
+        {
+            manager->sticky_keys_enabled(true);
+        }
+    }
+
+    void disable()
+    {
+        auto const locked_state = state.lock();
+        if (!locked_state->enabled)
+            return;
+
+        locked_state->enabled = false;
+        if (auto const manager = accessibility_manager.lock())
+        {
+            manager->sticky_keys_enabled(false);
+        }
+    }
+
+    void should_disable_if_two_keys_are_pressed_together(bool on)
+    {
+        if (auto const manager = accessibility_manager.lock())
+        {
+            manager->sticky_keys().should_disable_if_two_keys_are_pressed_together(on);
+        }
+        else
+        {
+            auto const locked_state = state.lock();
+            locked_state->should_disable_if_two_keys_are_pressed_together = on;
+        }
+    }
+
+    void on_modifier_clicked(std::function<void(int32_t)>&& callback)
+    {
+        if (auto const manager = accessibility_manager.lock())
+        {
+            manager->sticky_keys().on_modifier_clicked(std::move(callback));
+        }
+        else
+        {
+            auto const locked_state = state.lock();
+            locked_state->on_modifier_clicked = std::move(callback);
+        }
+    }
+
+private:
+    struct State
+    {
+        explicit State(bool enabled)
+            : enabled{enabled}
+        {}
+
+        bool enabled;
+        bool should_disable_if_two_keys_are_pressed_together = false;
+        std::function<void(int32_t)> on_modifier_clicked{[](int32_t) {}};
+    };
+
+    std::weak_ptr<msh::AccessibilityManager> accessibility_manager;
+    mir::Synchronised<State> state;
+};
+
+auto miral::StickyKeys::enabled() -> StickyKeys
+{
+    return StickyKeys{std::make_shared<Self>(true)};
+}
+
+auto miral::StickyKeys::disabled() -> StickyKeys
+{
+    return StickyKeys{std::make_shared<Self>(false)};
+}
+
+miral::StickyKeys::StickyKeys(live_config::Store& config_store)
+{
+    config_store.add_bool_attribute(
+        {"sticky_keys", "enable"},
+        "Whether or not sticky keys are enabled",
+        [this](live_config::Key const&, std::optional<bool> val)
```

Comment:
Would it be safer to capture `self`? `StickyKeys` is copyable and we may end up with a different instance surviving and a "dead" this pointer.